### ISSUE TITLE
Deprecation Warning: 'declaration-block-properties-order' has been deprecated and in 8.0

### ISSUE THREAD

**Issue by dangreen** [hudochenkov/stylelint-order#13]
With `order/declaration-block-property-groups-structure` rule stylelint prints next warning:
```
Deprecation Warning: 'declaration-block-properties-order'has been deprecated and in 8.0 will be removed. Instead use the community 'stylelint-order' plugin pack. See: https://stylelint.io/user-guide/rules/declaration-block-properties-order/
```

**Comment by dangreen** [comment]
I think you should update package on npm, because [this commit](https://github.com/hudochenkov/stylelint-order/commit/f1b7c8933074dcae05a3bd88bcc1a9c167725254) fixes this issue.

**Comment by hudochenkov** [comment]
Thank you for reporting! Please, update to `0.3.0`.

**Comment by bramwillemse** [comment]
I get the same warning with stylelint-order 0.4.3

**Comment by hudochenkov** [comment]
@bramwillemse you most likely have `declaration-block-properties-order` in your stylelint config. `stylelint-order` doesn't have any code that related to `declaration-block-properties-order`.

**Comment by bramwillemse** [comment]
@hudochenkov Thanks for the quick reply. I'm trying to find some examples how to implement the proper structure but can't really find it.. 

This is my current config: 
`
{
    "plugins": [
        "stylelint-order",
        "stylelint-selector-bem-pattern"
    ],
    "rules": {
        "order/declaration-block-property-groups-structure": [
            {
                "order": "strict",
                "properties": [
                    "display",
                    "visibility",
                    "float",
                    "position",
                    "top",
                    "right",
                    "bottom",
                    "left",
                    "flex",
                    "align",
                    "justify",
                    "vertical-align",
                    "width",
                    "height",
                    "min",
                    "max",
                    "overflow",
                    "margin",
                    "z-index"
                ]
            },
            {
                "emptyLineBefore": "always",
                "order": "strict",
                "properties": [
                    "appearance",
                    "padding",
                    "background",
                    "border",
                    "drop-shadow",
                    "outline"
                ]
            },
            {
                "emptyLineBefore": "always",
                "order": "strict",
                "properties": [
                    "content",
                    "font",
                    "font-family",
                    "src",
                    "font-style",
                    "font-size",
                    "font-weight",
                    "line-height",
                    "letter-spacing",
                    "color",
                    "white-space",
                    "text-align",
                    "text-shadow"
                ]
            },
            {
                "emptyLineBefore": "always",
                "order": "strict",
                "properties": [
                    "transform",
                    "translate",
                    "transition",
                    "animation"
                ]
            }
        ],
        "plugin/selector-bem-pattern": {
            "preset": "bem"
        },
        "at-rule-name-case": "lower",
        "at-rule-no-vendor-prefix": true,
        "block-closing-brace-empty-line-before": "never",
        "block-closing-brace-newline-after": "always",
        "block-closing-brace-newline-before": "always-multi-line",
        "block-no-empty": true,
        "color-hex-case": "upper",
        "color-named": "never",
        "color-no-invalid-hex": true,
        "comment-no-empty": true,
        "custom-property-empty-line-before": "always",
        "declaration-bang-space-after": "never",
        "declaration-bang-space-before": "always",
        "declaration-block-no-duplicate-properties": true,
        "declaration-block-no-redundant-longhand-properties": true,
        "declaration-block-no-shorthand-property-overrides": true,
        "declaration-block-semicolon-newline-after": "always",
        "declaration-block-semicolon-space-after": "always-single-line",
        "declaration-block-semicolon-space-before": "never",
        "declaration-block-trailing-semicolon": "always",
        "declaration-colon-space-after": "always",
        "declaration-colon-space-before": "never",
        "declaration-property-unit-blacklist": {},
        "declaration-property-unit-whitelist": {},
        "declaration-property-value-blacklist": {},
        "declaration-property-value-whitelist": {},
        "font-family-name-quotes": "always-where-recommended",
        "font-weight-notation": "numeric",
        "function-calc-no-unspaced-operator": true,
        "function-max-empty-lines": 5,
        "function-name-case": "lower",
        "function-url-quotes": "never",
        "function-whitespace-after": "always",
        "indentation": 4,
        "keyframe-declaration-no-important": true,
        "max-empty-lines": 5,
        "max-line-length": 150,
        "max-nesting-depth": 3,
        "media-feature-colon-space-after": "always",
        "media-feature-colon-space-before": "never",
        "media-feature-name-case": "lower",
        "media-feature-parentheses-space-inside": "always",
        "media-feature-range-operator-space-after": "always",
        "media-feature-range-operator-space-before": "always",
        "media-query-list-comma-newline-before": "never-multi-line",
        "media-query-list-comma-space-after": "always-single-line",
        "media-query-list-comma-space-before": "always-single-line",
        "no-duplicate-selectors": true,
        "no-empty-source": true,
        "no-extra-semicolons": true,
        "no-missing-end-of-source-newline": true,
        "no-unknown-animations": true,
        "number-leading-zero": "never",
        "number-max-precision": 3,
        "number-no-trailing-zeros": true,
        "property-case": "lower",
        "property-no-unknown": true,
        "selector-attribute-operator-space-after": "never",
        "selector-attribute-operator-space-before": "never",
        "selector-attribute-quotes": "always",
        "selector-combinator-space-after": "always",
        "selector-combinator-space-before": "always",
        "selector-descendant-combinator-no-non-space": true,
        "selector-list-comma-newline-after": "always-multi-line",
        "selector-list-comma-newline-before": "never-multi-line",
        "selector-list-comma-space-after": "never-single-line",
        "selector-list-comma-space-before": "never",
        "selector-max-empty-lines": 1,
        "selector-no-id": true,
        "selector-no-qualifying-type": [
            true,
            ignore: ["attribute", "class"]
        ],
        "selector-pseudo-class-case": "lower",
        "selector-pseudo-class-no-unknown": true,
        "selector-pseudo-element-case": "lower",
        "selector-pseudo-element-colon-notation": "single",
        "selector-pseudo-element-no-unknown": true,
        "selector-type-case": "lower",
        "selector-type-no-unknown": true,
        "shorthand-property-no-redundant-values": true,
        "string-no-newline": true,
        "string-quotes": "double",
        "unit-case": "lower",
        "unit-no-unknown": true,
        "value-list-comma-newline-after": "never-multi-line",
        "value-list-comma-newline-before": "never-multi-line",
        "value-list-comma-space-after": "always",
        "value-list-comma-space-before": "never",
        "value-list-max-empty-lines": 5,
        "value-no-vendor-prefix": true
    }
}
`

**Comment by hudochenkov** [comment]
@bramwillemse I believe, you can rename `order/declaration-block-property-groups-structure` to `order/properties-order` and it'll do the trick. Also you can safely remove `"order": "strict"`.

**Comment by bramwillemse** [comment]
@hudochenkov I've tried that, but doing that I get the following error:

`Error: Undefined rule order/properties-order
    at module.exports (<PROJECT>\node_modules\stylelint\lib\utils\configurationError.js:8:27)
    at <PROJECT>\node_modules\stylelint\lib\augmentConfig.js:279:13
    at Array.forEach (native)
    at normalizeAllRuleSettings (<PROJECT>\node_modules\stylelint\lib\augmentConfig.js:275:29)
    at <PROJECT>\node_modules\stylelint\lib\augmentConfig.js:84:12

`

I'm running gulp-stylelint.

**Comment by hudochenkov** [comment]
You aren't using `stylelint-order@0.4.3` then.

**Comment by bramwillemse** [comment]
Yes I am...

**Comment by bramwillemse** [comment]
Could it be an issue of gulp-stylelint using an older version of stylelint itself? I think I'll try and mitigate this by using postcss + stylelint on their own instead of using gulp-stylelint with all it's dependencies.

**Comment by hudochenkov** [comment]
@bramwillemse it shouldn't. You can try to remove `node_modules` and reinstall everything.

**Comment by bramwillemse** [comment]
@hudochenkov I already tried that before. To be sure I did it again, but that doesn't work.

**Comment by hudochenkov** [comment]
@bramwillemse could you create a repository with reproducible problem? You're gulpfile, package.json, stylelint config and one stylesheet.

**Comment by bramwillemse** [comment]
@hudochenkov Yeah, I'll try and do that. Might take a few days, since I have to get some work done first.. Thanks for your help!

**Comment by bramwillemse** [comment]
Okay, I just set the correct (latest) version numbers in my package.json and did another npm install. 
That took care of the deprecation warning. 

The weird thing, it didn't do anything else. Turns out, my colleague changed the gulp config and broke stylelint (a few months back already...). 

I don't really understand how 'npm view stylelint-order version' gave me the correct version number (same goes for gulp-stylelint) though.. 

Thanks for helping out! I'll have to dive into fixing the config. But at least we can call this progress :)

Cheers!
